Remove is_active filter while fetching users form es

AmitPhulera opened the pull request:
## Summary
<!--
    Provide a link to the ticket or document which prompted this change,
    Describe the rationale and design decisions.
-->
https://dimagi-dev.atlassian.net/browse/SAAS-11643
While downloading the mobile workers, the count that we were showing was for only active users but the downloaded users also contained inactive users also. Which resulted in the premature generation of a report download link when inactive users are more. And when the user used to click on the link it would result in 404.
## Feature Flag
<!-- If this is specific to a feature flag, which one? -->

## Product Description
<!-- For non-invisible changes, describe user-facing effects. -->

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below

### Automated test coverage

<!-- Identify the related test coverage and the tests it would catch -->

### QA Plan

<!--
- Describe QA plan that along with automated test coverages proves this PR is regression free
- Link to QA Ticket
-->

### Safety story
<!--
Describe any other pieces to the safety story including
local testing, why the change is inherently safe, and/or plans to limit the blast radius of a defect.
-->
Tested it locally, works fine. And is a small change that affects one place.
### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

AmitPhulera committed:
remove active filter while fetching users form es

mkangia commented on corehq/apps/users/dbaccessors/all_commcare_users.py:
```diff
@@ -56,7 +56,7 @@ def get_commcare_users_by_filters(domain, user_filters, count_only=False):
         return get_all_commcare_users_by_domain(domain)
 
-    query = _get_es_query(domain, user_filters)
+    query = _get_es_query(domain, user_filters).remove_default_filter('active')
```
confirming that you have seen all the usages of `get_commcare_users_by_filters`?
Also should we now rename it to `get_active_commcare_users_by_filters`?

AmitPhulera replied:
Yep yep. I have. It is only used in `bulk_download.py` and `users.py` which are serving this use-case only.
We are actually returning all the users, we have removed the default active filter.

mkangia replied:
So this change will effect only the count we fetch while downloading users?

AmitPhulera replied:
As I looked into the functioning we pass in `count_only` as True whenever we have to get count and that uses ES else to get user `get_all_commcare_users_by_domain` is called. So I guess it would be a yes.

mkangia commented:
Should we also have the same behavior for `bulk_download_usernames_async` ?

AmitPhulera commented:
> Should we also have the same behavior for bulk_download_usernames_async
Huh! I would have to look into it. I would revert after I have a look at that.

@mkangia

AmitPhulera commented:
Hey @mkangia, 
I looked into it, seems like `bulk_download_usernames_async` calls `dump_usernames` which calls `get_commcare_users_by_filters` [here](https://github.com/dimagi/commcare-hq/blob/ap%2Fugrade-to-couch-3/corehq/apps/users/bulk_download.py#L286) to get the count.

So I guess we are good here.

mkangia commented:
Hey @AmitPhulera 
But won't the user get the usernames for the inactive users as well but the count wont have that number?

AmitPhulera committed:
update get_mobile_usernames_by_filters to include inactive users also

AmitPhulera commented:
Hey @mkangia,
> But won't the user get the usernames for the inactive users as well but the count wont have that number?

This was the issue before, the PR addresses the exact issue you have mentioned.
```
_get_es_query(domain, user_filters).remove_default_filter('active')
```

Is the query that is generated to get the count. Now we have removed the default filter of `active` which would now include inactive users as well in the count.

But when I tried to download the usernames, I found out that when we download usernames inactive users are not included by default. 

Now I have updated `get_mobile_usernames_by_filters` and made the behavior common for downloading usernames and downloading the users. I have also verified the usage of this function, it is being used in one place only.

So now we would be including inactive users in the count, in username downloads and user downloads.

mkangia commented:
Cool. So can we just update `_get_es_query` itself instead of editing 2 places?

AmitPhulera committed:
update _get_es_query to include inactive users

AmitPhulera commented:
Done. 👍 
Thanks for the suggestion 😄

mkangia reviewed (approved):
Looks good.

Just pinging @dimagi/product for product perspective.
